File: java/InsertionSort.java
public class InsertionSort{


    public static void printMyArray(int [] myArray)
    {
        for(int i =0 ; i<myArray.length; i++)
        {
            System.out.print(myArray[i]+"\t");
        }
        System.out.println("\n");
    }

    public int [] sort(int [] arrToSort)
    {
        for(int i=1;i<arrToSort.length;i++)
        {
            int workingElement = arrToSort[i];
            System.out.println("working on key:\t"+arrToSort[i]+"\n");
            while(i>0 && workingElement < arrToSort[i-1])
            {
                arrToSort[i]=arrToSort[i-1];
                i = i-1;
            }
            arrToSort[i]=workingElement;
            printMyArray(arrToSort);
        }
       return arrToSort;

    }

    public static void main(String[] args)
    {

        InsertionSort insertionSort = new InsertionSort();

        int [] ArrayToBeSorted = {9,1,3,11,7,4,2,8,10,5,6};
        printMyArray(ArrayToBeSorted);

        int[] sortedArray = insertionSort.sort(ArrayToBeSorted);

        System.out.println("After Sorting");

        printMyArray(sortedArray);


    }
}

File: scala/InsertionSort.scala
object InsertionSort {

  def insertionSort(ArrayToSort:Array[Int]):Array[Int]={
    for(i<-1 until ArrayToSort.length)
    {
      var Until:Int = i;
      val workingElement:Int = ArrayToSort(i);
      println("working on element:\t"+workingElement+"\n");
      while(Until>0  && ArrayToSort(Until-1) > workingElement) {
          ArrayToSort(Until) = ArrayToSort(Until-1);
          Until = Until -1;
      }
      ArrayToSort(Until) = workingElement;
      println(ArrayToSort.mkString(","))
    }
    ArrayToSort;
  }

     def main(args: Array[String]): Unit = {

       val  myArray: Array[Int] = Array.apply(9, 1, 3, 11, 7, 4, 2, 8, 10, 5, 6);

       val sortedList = insertionSort(myArray);
       println("after sorting:\n")
       println(sortedList.mkString(","));
     }
}